firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java

```diff
@@ -165,6 +165,11 @@ private void recalculateAndSaveOverlays(Map<DocumentKey, MutableDocument> docs)
     // along the way.
     for (MutationBatch batch : batches) {
       for (DocumentKey key : batch.getKeys()) {
+        // If this batch has documents not included in passed in `docs`, skip them.
+        if (!docs.containsKey(key)) {
+          continue;
+        }
+
         FieldMask mask = masks.containsKey(key) ? masks.get(key) : FieldMask.EMPTY;
         mask = batch.applyToLocalView(docs.get(key), mask);
```

Comment:
Can we change this to only issue a single lookup in `docs` and use the nullable return value as the exit criteria?

Reply:
Done.